### PR TITLE
feat(app): report PAPI version during protocol analytics events

### LINKED ISSUE #4971: Protocol analytics: run app should collect Protocol API version
## overview

As an Opentrons engineer / project planner, I would like to understand how many of our users are still running Protocol API v1 so that I can:

- Understand what sort of pickup PAPIv2 is getting
- Figure out if/when support for PAPIv1 can be dropped to improve overall system reliability
- Drive further research and energy into increasing PAPIv2 usage

## acceptance criteria

- [ ] Protocol API version is reported with existing anonymous protocol data to mixpanel

### PR DESCRIPTION
**Pull request by mcous**
## overview

At some point, we're going to want to deprecate and remove version 1 of the Protocol API. It's buggy and its presence complicates / blocks a lot of fundamental software improvements we want to make.

This PR adds `protocolApiVersion` to the list of protocol event data so we can start to collect some PAPI usage metrics.

Closes #4971

## changelog

- feat(app): report PAPI version during protocol analytics events

## review requests

Turns out we didn't have any unit tests for the protocol analytics data, so I added some. Otherwise, the best way to see an e2e test of this will be to bug me at my desk because we don't have a lot of MixPanel logins to go around.

### CODE REVIEW

File: app/src/protocol/selectors.js [hunk 1]
```diff
@@ -181,6 +180,14 @@ export const getProtocolCreatorApp: CreatorAppSelector = createSelector(
   }
 )
 
+export const getProtocolApiVersion = (state: State): string | null => {
+  // TODO(mc, 2019-11-26): did not import due to circular dependency
+  // protocol API level should probably be part of protocol state
+  const level = state.robot.session.apiLevel
+
+  return level != null ? level.join('.') : null
```
Comment (amitlissack) [inline app/src/protocol/selectors.js]: below you used `apiVersion !== null` and here you used `level != null`. Is that intentional?

Comment (mcous) [reply app/src/protocol/selectors.js]: Yeah, the `apiVersion` below is coming out of this function here (`getProtocolApiVersion`), which I know will mean the variable be one of `string` or `null`

`state.robot.session.apiLevel` is a value that is set by the RPC server/client, and that code is not type-checked on the JS side and just generally needs a little more caution. The `!=` here is to tolerate `state.robot.session.apiLevel` being `undefined` (e.g. missing entirely) and treat it the same as if it were set properly to `null`

Comment (iansolano) [reply app/src/protocol/selectors.js]: can we leave this `the != here is to tolerate state.robot.session.apiLevel being undefined` as a comment

**Comment by mcous** [comment]
Synced up with edge and ran some manual e2e tests and it looks like it's all working

**Pull request merged**